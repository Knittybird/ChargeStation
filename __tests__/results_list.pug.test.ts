

const pug = require('pug');

const chargers = [
  {
    address: {
      addressLine: "222 SW Harrison Street",
      title: "Harrison Tower",
      town: "Portland",
      state: "OR",
      postalCode: "97201"
    },
    pos: {
      lat: 45.5097989,
      lng: -122.6794873
    }
  },
  {
    address: {
      addressLine: "100 SW Market St",
      title: "Park Square",
      town: "Portland",
      state: "OR",
      postalCode: "97201"
    },
    pos: {
      lat: 45.510792,
      lng: -122.67764
    }
  },
  {
    address: {
      addressLine: "1510 SW Harbor Wy",
      title: "Kimpton Riverplace Hotel",
      town: "Portland",
      state: "OR",
      postalCode: "97201"
    },
    pos: {
      lat: 45.511486,
      lng: -122.674101
    }
  },
  {
    address: {
      addressLine: "200 SW Market St, Suite 1720",
      title: "200 SW Market Building",
      town: "Portland",
      state: "OR",
      postalCode: "97201"
    },
    pos: {
      lat: 45.511592,
      lng: -122.6786039 
    },
  },
  {
    address: {
      addressLine: "1945 SE Water Ave",
      title: "Oregon Museum of Science and Industry (OMSI)",
      town: "Portland",
      state: "OR",
      postalCode: "97214"
    },
    pos: {
      lat: 45.508501,
      lng: -122.665932
    }
  },
  {
    address: {
      addressLine: "1401 Southwest Naito Parkway",
      title: "Portland Marriott Downtown Waterfront",
      town: "Portland",
      state: "OR",
      postalCode: "97201"
    },
    pos: {
      lat: 45.5123866,
      lng: -122.6756475
    }
  },
  {
    address: {
      addressLine: "2145 SE Water Avenue",
      title: "2145 SE Water Avenue",
      town: "Portland",
      state: "OR",
      postalCode: "97214"
    },
    pos: {
      lat: 45.50932,
      lng: -122.66639
    }
  }
]

const renderPug = data => pug.renderFile('views/results_list.pug', data);

test('It should render pug correctly', () => {
  expect(renderPug(chargers)).toMatchSnapshot();
});