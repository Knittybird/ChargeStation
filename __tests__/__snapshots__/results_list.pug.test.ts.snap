// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render pug correctly 1`] = `"<!DOCTYPE html><html><meta name=\\"viewport\\" content=\\"initial-scale=1.0, user-scalable=no\\"><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.4.1/css/bootstrap.min.css\\"><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/style.css\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta charset=\\"utf-8\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"><title>Where's a Charger</title></html><body class=\\"bg-success\\"><nav class=\\"navbar navbar-expand-md navbar-dark bg-dark py-1\\"><a class=\\"navbar-brand\\" href=\\"index\\">Where&apos;s A Charger?</a><button class=\\"navbar-toggler\\" type=\\"button\\" data-toggle=\\"collapse\\" data-target=\\"#navbarNavAltMarkup\\" aria-controls=\\"navbarNavAltMarkup\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"collapse navbar-collapse\\" id=\\"navbarNavAltMarkup\\"><div class=\\"navbar-nav nav\\"><a class=\\"nav-item nav-link\\" href=\\"index\\">Home </a><a class=\\"nav-item nav-link\\" href=\\"about\\">About</a><a class=\\"nav-item nav-link\\" href=\\"search\\">Search</a><a class=\\"nav-item nav-link\\" href=\\"directions\\">Directions</a></div></div></nav><div class=\\"container\\"><h1>Results</h1><table class=\\"table-responsive table-hover\\" id=\\"chargers\\"><thead><tr><th>Name</th><th>Address</th><th></th><th class=\\"d-none\\">Lattitude, Longitude</th></tr></thead><tbody><tr><td>Harrison Tower</td><td>222 SW Harrison Street Portland, OR 97201</td><td class=\\"d-none\\">45.5097989, -122.6794873</td></tr><tr><td>Park Square</td><td>100 SW Market St Portland, OR 97201</td><td class=\\"d-none\\">45.510792, -122.67764</td></tr><tr><td>Kimpton Riverplace Hotel</td><td>1510 SW Harbor Wy Portland, OR 97201</td><td class=\\"d-none\\">45.511486, -122.674101</td></tr><tr><td>200 SW Market Building</td><td>200 SW Market St, Suite 1720 Portland, OR 97201</td><td class=\\"d-none\\">45.511592, -122.6786039</td></tr><tr><td>Oregon Museum of Science and Industry (OMSI)</td><td>1945 SE Water Ave Portland, OR 97214</td><td class=\\"d-none\\">45.508501, -122.665932</td></tr><tr><td>Portland Marriott Downtown Waterfront</td><td>1401 Southwest Naito Parkway Portland, OR 97201</td><td class=\\"d-none\\">45.5123866, -122.6756475</td></tr><tr><td>2145 SE Water Avenue</td><td>2145 SE Water Avenue Portland, OR 97214</td><td class=\\"d-none\\">45.50932, -122.66639</td></tr></tbody></table></div></body><script src=\\"https://code.jquery.com/jquery-3.4.1.slim.min.js\\" integrity=\\"sha384-J6qa4849blE2+poT4WnyKhv5vZF5SrPo0iEjwBvKU7imGFAV0wwj1yYfoRSJoZ+n\\" crossorigin=\\"anonymous\\"></script><script src=\\"https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js\\" integrity=\\"sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo\\" crossorigin=\\"anonymous\\"></script><script src=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js\\" integrity=\\"sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6\\" crossorigin=\\"anonymous\\"></script>"`;
